miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/MenuController.java

```diff
@@ -154,4 +192,166 @@ public ModelAndView tissueOptions(ModelMap model) {
   public void setServletContext(ServletContext servletContext) {
     this.servletContext = servletContext;
   }
+
+  abstract static class Fetch<T> {
+    protected final String name;
+
+    Fetch(String name) {
+      this.name = name;
+    }
+
+    public abstract void write(ObjectNode node, T item);
+
+    static <T> Fetch<T> forBool(String name, Predicate<T> getter) {
+      return new Fetch<T>(name) {
+
+        @Override
+        public void write(ObjectNode node, T item) {
+          node.put(name, getter.test(item));
+        }
+      };
+    }
+
+    static <T> Fetch<T> forInt(String name, ToIntFunction<T> getter) {
+      return new Fetch<T>(name) {
+        @Override
+        public void write(ObjectNode node, T item) {
+          node.put(name, getter.applyAsInt(item));
+        }
+      };
+    }
+
+    static <T> Fetch<T> forLong(String name, ToLongFunction<T> getter) {
+      return new Fetch<T>(name) {
+
+        @Override
+        public void write(ObjectNode node, T item) {
+          node.put(name, getter.applyAsLong(item));
+        }
+
+      };
+    }
+
+    static <T, E extends Enum<E>> Fetch<T> forEnum(String name, Function<T, E> getter) {
+      return new Fetch<T>(name) {
+
+        @Override
+        public void write(ObjectNode node, T item) {
+          E value = getter.apply(item);
+          node.put(name, value == null ? null : value.name());
+        }
+
+      };
+    }
+
+    static <T> Fetch<T> forString(String name, Function<T, String> getter) {
+      return new Fetch<T>(name) {
+
+        @Override
+        public void write(ObjectNode node, T item) {
+          node.put(name, getter.apply(item));
+        }
+
+      };
+    }
+
+    @SafeVarargs
+    static <T, S> Fetch<T> nested(String name, Function<T, Iterable<S>> getter, Fetch<S>... writers) {
+      return new Fetch<T>(name) {
+
+        @Override
+        public void write(ObjectNode node, T item) {
+          ArrayNode array = node.putArray(name);
+          for (S child : getter.apply(item)) {
+            ObjectNode childObject = array.addObject();
+            for (Fetch<S> writer : writers) {
+              writer.write(childObject, child);
+            }
+          }
+        }
+      };
+    }
+  }
+
+  @SafeVarargs
+  private static <T> void createArray(ObjectNode node, String name, Iterable<T> items, Fetch<T>... writers) {
+    ArrayNode array = node.putArray(name);
+    for (T item : items) {
+      ObjectNode dto = array.addObject();
+      for (Fetch<T> writer : writers) {
+        writer.write(dto, item);
+      }
+    }
+  }
+
+  @RequestMapping(path = "/constants.js")
+  @ResponseBody
+  public String constantsScript(HttpServletResponse response) throws IOException {
```

Comment:
Oh, I didn't notice DTOs were used for some. It would be nicer to create the missing DTOs. If you'd rather merge as is, can we call this temporary and make a ticket for creating and using DTOs for consistency?